android/quest/src/main/java/org/smartregister/fhircore/quest/util/PatientUtil.kt

```diff
@@ -0,0 +1,124 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.util
+
+import ca.uhn.fhir.rest.gclient.ReferenceClientParam
+import ca.uhn.fhir.rest.gclient.TokenClientParam
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.Search
+import com.google.android.fhir.search.search
+import org.hl7.fhir.r4.model.CodeableConcept
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Condition
+import org.hl7.fhir.r4.model.Enumerations
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.ResourceType
+import org.smartregister.fhircore.engine.configuration.ConfigurationRegistry
+import org.smartregister.fhircore.quest.configuration.view.Code
+import org.smartregister.fhircore.quest.configuration.view.DynamicColor
+import org.smartregister.fhircore.quest.configuration.view.Filter
+import org.smartregister.fhircore.quest.configuration.view.PatientRegisterRowViewConfiguration
+import org.smartregister.fhircore.quest.configuration.view.Properties
+import org.smartregister.fhircore.quest.configuration.view.Property
+import org.smartregister.fhircore.quest.data.patient.model.AdditionalData
+
+suspend fun loadAdditionalData(
+  patientId: String,
+  configurationRegistry: ConfigurationRegistry,
+  fhirEngine: FhirEngine
+): List<AdditionalData> {
+  val result = mutableListOf<AdditionalData>()
+
+  val patientRegisterRowViewConfiguration =
+    configurationRegistry.retrieveConfiguration<PatientRegisterRowViewConfiguration>(
+      configClassification = QuestConfigClassification.PATIENT_REGISTER_ROW
+    )
+
+  patientRegisterRowViewConfiguration.filters?.forEach { filter ->
+    when (filter.resourceType) {
+      Enumerations.ResourceType.CONDITION -> {
+        val conditions =
+          getSearchResults<Condition>(patientId, Condition.SUBJECT, filter, fhirEngine)
+
+        val sortedByDescending = conditions.maxByOrNull { it.recordedDate }
+        sortedByDescending?.category?.forEach { cc ->
+          cc.coding.firstOrNull { c -> c.code == filter.valueCoding!!.code }?.let {
+            val status = sortedByDescending.code?.coding?.firstOrNull()?.display ?: ""
+            result.add(
+              AdditionalData(
+                label = filter.label,
+                value = status,
+                valuePrefix = filter.valuePrefix,
+                properties = propertiesMapping(status, filter)
+              )
+            )
+          }
+        }
+      }
+    }
+  }
+
+  return result
+}
+
+private suspend inline fun <reified T : Resource> getSearchResults(
+  patientId: String,
+  reference: ReferenceClientParam,
+  filter: Filter,
+  fhirEngine: FhirEngine
+): List<T> {
+  return fhirEngine.search {
+    filterByPatient(reference, patientId)
+
+    when (filter.valueType) {
```

Comment:
Add missing branches or else branch for `when` expression on `enum`

Reply:
added `else` branch